### PR TITLE
[docs] MongoDB auto-user provisioning

### PR DESCRIPTION
**Pull request by greedy52**
Related:
- https://github.com/gravitational/teleport/pull/33717

Changes:
- Fixed release number for MariaDB/MySQL auto-user provisioning (already fixed on v14 release branch)
- Added new guide for MongoDB auto-user provisioning
- Broke `common-teleport-role.mdx` to smaller includes

### CODE REVIEW

**Review by Tener — APPROVED** [review]

File: docs/pages/database-access/auto-user-provisioning/mongodb.mdx [hunk 1]
```diff
@@ -0,0 +1,121 @@
+---
+title: MongoDB Automatic User Provisioning 
+description: Configure automatic user provisioning for MongoDB.
+---
+
+(!docs/pages/includes/database-access/auto-user-provisioning/intro.mdx!)
+
+- Teleport cluster v14.3 or above with a configured [self-hosted
+  MongoDB](../guides/mongodb-self-hosted.mdx).
```
Comment (ptgott) [inline docs/pages/database-access/auto-user-provisioning/mongodb.mdx:9]:
```suggestion
- Teleport cluster v14.3 or above.
- A self-hosted
  MongoDB database enrolled with your Teleport cluster. Follow the [Teleport documentation](../guides/mongodb-self-hosted.mdx) to learn how to enroll your database.
```
Suggested to be more explicit about what "with a configured..." means and what the link points to.

File: docs/pages/database-access/auto-user-provisioning/mongodb.mdx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+Teleport uses the same authentication mechanism (X.509) when connecting as an
+admin user as for regular user connections.
+
+The admin user must have privileges within the database to create users and
+grant them privileges. The admin user must also have privileges to monitor user
+connections.
+
+First create a role on `admin` database with the following privileges:
+```bash
```
Comment (ptgott) [inline docs/pages/database-access/auto-user-provisioning/mongodb.mdx:26]:
```suggestion
First, create a role on the `admin` database with the following privileges:

```bash
```

File: docs/pages/database-access/auto-user-provisioning/mongodb.mdx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    privileges: [
+        { resource: { cluster: true }, actions: [ "inprog" ] },
+        { resource: { db: "", collection: "" }, actions: [ "grantRole", "revokeRole" ] },
+        { resource: { db: "$external", "collection": "" }, actions: [ "createUser", "updateUser", "dropUser", "viewUser", "setAuthenticationRestriction", "changeCustomData"] },
+    ],
+    roles: [],
+})
+```
+
+<Details title="Limit grantRole action to specific databases">
```
Comment (ptgott) [inline docs/pages/database-access/auto-user-provisioning/mongodb.mdx:38]:
```suggestion
<Details title="Limit the grantRole action to specific databases">
```

File: docs/pages/database-access/auto-user-provisioning/mongodb.mdx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        { resource: { db: "<db1>", collection: "" }, actions: [ "grantRole" ] },
+        { resource: { db: "<db2>", collection: "" }, actions: [ "grantRole" ] },
+        ...
+    ],
+    roles: [],
+})
+```
+</Details>
+
+Now create the admin user with this role:
```
Comment (ptgott) [inline docs/pages/database-access/auto-user-provisioning/mongodb.mdx:63]:
```suggestion
Now create the admin user with this role:

```

File: docs/pages/database-access/auto-user-provisioning/mongodb.mdx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+db.getSiblingDB("$external").runCommand({
+  createUser: "CN=teleport-admin",
+  roles: [ { role: 'teleport-admin-role', db: 'admin' } ],
+})
+```
+
+(!docs/pages/includes/database-access/auto-user-provisioning/db-definition-self-hosted.mdx protocol="mongodb" uri="localhost:27017"!)
+
+
+## Step 2/3. Configure Teleport role
```
Comment (ptgott) [inline docs/pages/database-access/auto-user-provisioning/mongodb.mdx:74]:
```suggestion
## Step 2/3. Configure a Teleport role
```

File: docs/pages/database-access/auto-user-provisioning/mongodb.mdx [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+    - "{{external.db_roles}}"
+```
+
+(!docs/pages/includes/database-access/auto-user-provisioning/db_users_ignored.mdx!)
+
+(!docs/pages/includes/database-access/auto-user-provisioning/modes.mdx!)
+
+Users created within the database will:
+
+- Have the same name as Teleport username.
```
Comment (ptgott) [inline docs/pages/database-access/auto-user-provisioning/mongodb.mdx:108]:
```suggestion
- Have the same username as the authenticated Teleport user.
```

File: docs/pages/database-access/auto-user-provisioning/mongodb.mdx [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+- Have `teleport-auto-user` set to `true` in user's `customData`.
```
Comment (ptgott) [inline docs/pages/database-access/auto-user-provisioning/mongodb.mdx:109]:
```suggestion
- Have `teleport-auto-user` set to `true` in the user's `customData`.
```

**Review by ptgott — APPROVED** [review]
Approved with suggestions

**Pull request merged**